src/fu-engine.c

```diff
@@ -6031,6 +6031,19 @@ fu_engine_add_device(FuEngine *self, FuDevice *device)
 			  fu_device_get_name(device));
 	}
 
+#ifndef SUPPORTED_BUILD
+	/* we don't know if this device has a signed or unsigned payload */
+	if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE) &&
+	    !fu_device_has_flag(device, FWUPD_DEVICE_FLAG_SIGNED_PAYLOAD) &&
+	    !fu_device_has_flag(device, FWUPD_DEVICE_FLAG_UNSIGNED_PAYLOAD) &&
+	    !fu_device_has_internal_flag(device, FU_DEVICE_INTERNAL_FLAG_MD_SET_SIGNED)) {
+		g_warning("%s device does not define payload %s [%s]",
```

Comment:
> warning shows up during development

Yes, that was my intention. I was going for irritation rather than asserting as I'm still waiting for ~20% of the vendors to respond with the data.

Reply:
OK.